### PR TITLE
[core] add safestate unit test, demonstrate two bugs

### PR DESCRIPTION
**Pull request by knopers8**

### CODE REVIEW

File: core/workflow/safestate_test.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+				subroles[1] = &taskRole{
+					roleBase: roleBase{state: SafeState{state: task.CONFIGURED}},
+					Traits:   task.Traits{Trigger: "", Await: "", Timeout: "", Critical: true},
+				}
+				ar := &aggregatorRole{
+					roleBase:   roleBase{state: SafeState{state: task.CONFIGURED}},
+					aggregator: aggregator{subroles},
+				}
+				ar.state.merge(task.ERROR, ar)
+				Expect(ar.GetState()).To(Equal(task.CONFIGURED)) // fixme: this fails
```
Comment (knopers8) [inline core/workflow/safestate_test.go:130]: This seems to be related to https://its.cern.ch/jira/browse/OCTRL-853

File: core/workflow/safestate_test.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+				subroles[1] = &taskRole{
+					roleBase: roleBase{state: SafeState{state: task.DONE}},
+					Traits:   task.Traits{Trigger: "", Await: "", Timeout: "", Critical: true},
+				}
+				ar := &aggregatorRole{
+					roleBase:   roleBase{state: SafeState{state: task.RUNNING}},
+					aggregator: aggregator{subroles},
+				}
+				ar.state.merge(task.DONE, ar)
+				Expect(ar.GetState()).To(Equal(task.MIXED))
```
Comment (knopers8) [inline core/workflow/safestate_test.go:149]: This behaves as expected now, but according to the requirements stated in https://its.cern.ch/jira/browse/OCTRL-846, the aggregated state should go to ERROR as long as the expected state is RUNNING. I think we will have to introduce a new concept of an expected state for the tasks...

File: core/workflow/safestate_test.go [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+					roleBase: roleBase{state: SafeState{state: task.MIXED}},
+					Traits:   task.Traits{Trigger: "", Await: "", Timeout: "", Critical: true},
+				}
+				ar := &aggregatorRole{
+					roleBase:   roleBase{state: SafeState{state: task.ERROR}},
+					aggregator: aggregator{subroles},
+				}
+				ar.state.merge(task.MIXED, ar)
+				Expect(ar.GetState()).To(Equal(task.ERROR)) // FIXME: this fails
+			})
```
Comment (knopers8) [inline core/workflow/safestate_test.go:62]: @teo may i get your confirmation that this test case should not fail?

File: core/workflow/safestate_test.go (hunk 1 above)
Comment (teo) [reply core/workflow/safestate_test.go:160]: @justonedev1 ↑

File: core/workflow/safestate_test.go (hunk 3 above)
Comment (teo) [reply core/workflow/safestate_test.go]: Correct. `ERROR`×`ANYTHING` is still `ERROR`.

**Review by teo — APPROVED** [review]

**Pull request merged**